libs/common-services/src/main/java/uk/gov/di/ipv/core/library/persistence/item/ClientOAuthSessionItem.java

```diff
@@ -41,9 +45,46 @@ public List<String> getScopeClaims() {
         return Arrays.asList(this.scope.split(" "));
     }
 
-    public List<Vot> getRequestedVotsByStrength() {
+    public List<Vot> getRequestedVotsByStrengthDescending() {
         return Vot.SUPPORTED_VOTS_BY_DESCENDING_STRENGTH.stream()
                 .filter(vot -> vtr.contains(vot.name()))
                 .toList();
     }
+
+    // Refactor this out in PYIC-6984
+    public Vot getLowestStrengthRequestedGpg45Vot(ConfigService configService) {
+        var requestedGpg45VotsByStrengthDescending =
+                getRequestedVotsByStrengthDescending().stream()
+                        .filter(vot -> vot.getProfileType() == ProfileType.GPG45)
+                        .toList();
+
+        var lowestStrengthRequestedGpg45Vot =
+                requestedGpg45VotsByStrengthDescending.get(
+                        requestedGpg45VotsByStrengthDescending.size() - 1);
```

Comment:
Do we want to check these indexes exist?

Reply:
This shouldn't ever be a problem and as you say if it does cause an issue we will know about it from the out of bounds exception that gets thrown. We could add more code here, but I think all it would get us is a nicer error message in the logs (which is often worthwhile for errors that we think might happen).